openSUSE-2017.7.2/salt.spec

```diff
@@ -448,7 +577,22 @@ cd doc && make html && rm _build/html/.buildinfo && rm _build/html/_images/proxy
 %endif
 
 %install
+%if 0%{?build_py2} && 0%{?default_py3}
+mv _build.python2 build
+%{__python} setup.py --salt-transport=both install --prefix=%{_prefix} --root=%{buildroot}
+mv build _build.python2
+%endif
+%if 0%{?build_py3}
+mv _build.python3 build
 %{__python3} setup.py --salt-transport=both install --prefix=%{_prefix} --root=%{buildroot}
+mv build _build.python3
+%endif
+%if 0%{?build_py2} && !0%{?default_py3}
+mv _build.python2 build
+rm -f %{buildroot}%{_bindir}/*
+%{__python} setup.py --salt-transport=both install --prefix=%{_prefix} --root=%{buildroot}
+mv build _build.python2
+%endif
```

Comment:
Would it be easier to understand if we would refactor the logic like this?

```
if default-py3:
  if build-py2:
    execute py2-setup
    remove bindir
  if build-py3:
    execute py3-setup
else:
  if build-py3:
    execute py3-setup
    remove bindir
  if build-py2:
    execute py2-setup
```

Reply:
As discussed with @dincamihai , I agree, this refactor makes easier to understand the logic behind. So 👍

Reply:
yes, but I would like to see only one time py2-setup and py3-setup at all and just provide an
info which should install the scripts.

Or a special call at the end which only install the scripts.

Would be good if something like this would be possible with `setup.py`